kafka-client/src/main/java/dev/responsive/kafka/store/ResponsiveStateListener.java

```diff
@@ -0,0 +1,65 @@
+package dev.responsive.kafka.store;
+
+import java.io.Closeable;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.Gauge;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.streams.KafkaStreams.State;
+import org.apache.kafka.streams.KafkaStreams.StateListener;
+
+public class ResponsiveStateListener implements StateListener, Closeable {
+  private final Metrics metrics;
+  private final MetricName stateMetricName;
+
+  private StateListener userStateListener;
+  private State currentState;
+
+  /**
+   * @param applicationId the Streams application id, ie the shared consumer group id
+   * @param streamsClientId the Streams client id, ie the process-specific id
+   */
+  public ResponsiveStateListener(
+      final Metrics metrics,
+      final String applicationId,
+      final String streamsClientId
+  ) {
+    this.metrics = metrics;
+    final Map<String, String> tags = new HashMap<>();
+    tags.put("application-id", applicationId);
+    tags.put("client-id", streamsClientId);
+
+    stateMetricName = new MetricName(
+        "state",
+        "stream-application-metrics",
```

Comment:
for other streams metrics it seems we used `responsive.streams` - any reason not to use that group here?

Reply:
Well right now at least the only metrics with that seem to be topic metrics that we added -- personally, I would've just put those in the `"stream-topic-metrics"` group with the other built-in topic metrics in Streams

The idea being to keep metrics scoped to the same level/tags in the same group -- but that's just my random personal philosophy, if there's a better way to organize things I'll go along. We can keep all the responsive metrics in the same group, it's just a bit awkward when we have metrics that are really streams metrics...

that's the long answer. Short answer is, anything works